### PR TITLE
Update docs wrt auth-v2 feature

### PR DESCRIPTION
**Pull request by nuivall**
This CL updates documentation with regards to auth-v2 feature.

Fixes https://github.com/scylladb/scylladb/issues/17061
Fixes https://github.com/scylladb/scylladb/issues/17718

### CODE REVIEW

File: docs/troubleshooting/password-reset.rst [hunk 1]
```diff
@@ -13,11 +13,11 @@ Procedure
 
    sudo systemctl stop scylla-server
 
-| 2. Remove your tables under ``/var/lib/scylla/data/system_auth/``.
+| 2. Remove your tables under ``/var/lib/scylla/data/system_auth_v2/``.
```
Comment (nuivall) [inline docs/troubleshooting/password-reset.rst:16]: @kbr-scylla will this work when raft is involved or we need to do something more?

File: docs/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/enable-consistent-topology.rst [hunk 2]
```diff
@@ -57,6 +57,9 @@ In particular, you must abstain from:
 * :doc:`Cluster management procedures </operating-scylla/procedures/cluster-management/index>` (adding, replacing, removing, decommissioning nodes etc.).
 * Running :doc:`nodetool repair </operating-scylla/nodetool-commands/repair>`.
 * Running :doc:`nodetool checkAndRepairCdcStreams </operating-scylla/nodetool-commands/checkandrepaircdcstreams>`.
+* Any modifications of :doc:`authentication </operating-scylla/security/authentication>` and :doc:`authorization </operating-scylla/security/enable-authorization>` settings.
+* Any change of authorization via :doc:`CQL API </operating-scylla/security/authorization>`.
+* Doing schema changes.
```
Comment (nuivall) [inline docs/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/enable-consistent-topology.rst:62]: technically not all schema changes but to simplify I mention all (more info in commit description)

**Comment by ptrsmrn** [comment]
We do not mention:
1. what's the reason behind introducing auth v2 - we could provide some historical background, a short paragraph should be enough
2. what's going to happen with auth v1 - actually, wha't the plan here? I presume it's deprecated and will going to be obsoleted?
3. we're still supporting both tables (auth v2 with `consistent-topology-changes` and auth v1 without), but the doc removes auth v1 mentions
4. what's the up-/downgrade procedure or how the data migration from auth v1 to v2 and vice versa looks like

**Comment by nuivall** [comment]
> We do not mention:
> 
> 1. what's the reason behind introducing auth v2 - we could provide some historical background, a short paragraph should be enough

feature is transparent for the user so why write something like this?  we can write a blog post instead. Also I think docs are just for documenting what is, not what was and the history of the product.

> 2. what's going to happen with auth v1 - actually, wha't the plan here? I presume it's deprecated and will going to be obsoleted?

yes

> 3. we're still supporting both tables (auth v2 with `consistent-topology-changes` and auth v1 without), but the doc removes auth v1 mentions

no idea how we want to show it, it complicates the docs of many procedures to mention that for v1 do this for v2 don't do this

> 4. what's the up-/downgrade procedure or how the data migration from auth v1 to v2 and vice versa looks like

there is no downgrade procedure

**Comment by ptrsmrn** [comment]
> > We do not mention:
> > 
> > 1. what's the reason behind introducing auth v2 - we could provide some historical background, a short paragraph should be enough
> 
> feature is transparent for the user so why write something like this? we can write a blog post instead. Also I think docs are just for documenting what is, not what was and the history of the product.

Not sure if it's transparent for all the users - maybe. I mentioned audit schema migration in the docs - https://github.com/scylladb/scylla-enterprise/pull/3752/files#diff-263671934a9bf720a50f2e77cfd61e5bcdf1dd95b90919eddfb4d8263b60e619R151
My thinking was that users may be surprised by auth v2 and ask what was the v1 then.

> > 2. what's going to happen with auth v1 - actually, wha't the plan here? I presume it's deprecated and will going to be obsoleted?
> 
> yes

Worth mentioning, especially if we have a rough estimate on when v1 is going to be removed.

> > 3. we're still supporting both tables (auth v2 with `consistent-topology-changes` and auth v1 without), but the doc removes auth v1 mentions
> 
> no idea how we want to show it, it complicates the docs of many procedures to mention that for v1 do this for v2 don't do this

Maybe if you begin the doc by stating that there's auth v1 working without `consistent-topology-changes` and auth v2 working with it, you can write `system_auth/system_auth_v2` for cases that work for both and leave only v1 or v2 for cases that work only for specific version.

> > 4. what's the up-/downgrade procedure or how the data migration from auth v1 to v2 and vice versa looks like
> 
> there is no downgrade procedure

So it's worth mentioning there's no downgrade procedure

**Comment by nuivall** [comment]
> This is a great documentation update. I'll approve this PR, but I want to make sure I completely understand the change. @nuivall, is there any user action required on existing clusters? If not, this PR is good as is.
> 
> But you're discussing the upgrade/downgrade procedure above, so I'm not sure.
> 
> * If the user has to do anything to upgrade to `system_auth_v2`, we should document the upgrade procedure and add a note that there's no downgrade procedure.
> * If the user doesn't have to do anything to upgrade, there's no point in mentioning the lack of a downgrade procedure.

It's connected to consistent topology upgrade procedure (does it how a downgrade?) so user need to perform it but nothing else. That's for existing clusters. For new ones consistent topology and therefore auth-v2 are auto enabled.


> * In new clusters, consistent topology changes are enabled, so the user needs to use `system_auth_v2` - is that correct?

Yes but they don't use it directly, there is separate CQL statement syntax (e.g. CREATE ROLE) without specifying keyspace name.

> * In existing clusters, if you upgrade to 6.0 and manually enable consistent topology changes, you need to use `system_auth_v2`, but if you upgrade without enabling consistent topology changes, you should use `system_auth `- is this how it works?

Yes, my understanding is that you have to enable consistent topology changes to fully perform the upgrade. We've worded it like this in the procedure: ` If you have such a cluster, then you need to enable consistent topology changes `

Altough perhaps 5.4->6.0 page is missing something? It doesn't link to general upgrade procedure
http://previews.docs.scylladb.com/scylladb/17951/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-6.0/ something similar to previous one: http://previews.docs.scylladb.com/scylladb/17951/upgrade/upgrade-opensource/upgrade-guide-from-5.4-to-5.5/upgrade-guide-from-5.4-to-5.5-generic/

**Comment by nuivall** [comment]
> > > We do not mention:
> > > 
> > > 1. what's the reason behind introducing auth v2 - we could provide some historical background, a short paragraph should be enough
> > 
> > 
> > feature is transparent for the user so why write something like this? we can write a blog post instead. Also I think docs are just for documenting what is, not what was and the history of the product.
> 
> Not sure if it's transparent for all the users - maybe. I mentioned audit schema migration in the docs - https://github.com/scylladb/scylla-enterprise/pull/3752/files#diff-263671934a9bf720a50f2e77cfd61e5bcdf1dd95b90919eddfb4d8263b60e619R151 My thinking was that users may be surprised by auth v2 and ask what was the v1 then.

That's different, audit log is something read by users, and it's not an internal system table.

> 
> > > 2. what's going to happen with auth v1 - actually, wha't the plan here? I presume it's deprecated and will going to be obsoleted?
> > 
> > 
> > yes
> 
> Worth mentioning, especially if we have a rough estimate on when v1 is going to be removed.
> 

We don't and I don't see why it would be worth to mention. It's internal thing, if you have properly written application you don't need to change anything.

> > > 3. we're still supporting both tables (auth v2 with `consistent-topology-changes` and auth v1 without), but the doc removes auth v1 mentions
> > 
> > 
> > no idea how we want to show it, it complicates the docs of many procedures to mention that for v1 do this for v2 don't do this
> 
> Maybe if you begin the doc by stating that there's auth v1 working without `consistent-topology-changes` and auth v2 working with it, you can write `system_auth/system_auth_v2` for cases that work for both and leave only v1 or v2 for cases that work only for specific version.
> 
> > > 4. what's the up-/downgrade procedure or how the data migration from auth v1 to v2 and vice versa looks like
> > 
> > 
> > there is no downgrade procedure
> 
> So it's worth mentioning there's no downgrade procedure

I don't see anything about downgrade of an upgrade (dunno it it is even officially supported). So if we were not mentioning this before I don't see it related to this PR.

**Comment by tzach** [comment]
Why is auth-v2 coupled with a safe topology update?
I understand both use Raft, but are they dependent on each other?

**Comment by nuivall** [comment]
> Why is auth-v2 coupled with a safe topology update? I understand both use Raft, but are they dependent on each other?

Mainly because it doesn't have it's own migration procedure it taps to consistent topology upgrade procedure.

**Pull request closed without merging**